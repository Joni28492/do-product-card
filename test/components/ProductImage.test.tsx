import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/prodcuts';


describe('ProductImage', () => {


    test('Debe de mostrar el componente correctamente con la iamagen personalizado', () => {
        const wrapper = renderer.create(
            <ProductImage img="https://hola.jpg" />
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    })

    test('debe mostrar el componente con la imagen del producto', () => {


        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (<ProductImage />)
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();//u para actualizar snapshot
    })


})