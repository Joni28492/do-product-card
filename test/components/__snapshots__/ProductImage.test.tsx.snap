// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente correctamente con la iamagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`ProductImage debe mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
